client/extensions/woocommerce/app/settings/payments/payments-location-currency.js

```diff
@@ -41,11 +68,38 @@ class SettingsPaymentsLocationCurrency extends Component {
 				<Card>
 					<AddressView
 						address={ this.state.address } />
+
+					<FormSelect className="payments__currency-select" value={ currency.value }>
+						{
+							currency.options &&
+							Object.keys( currency.options ).map(
+								( o ) => this.renderOption( o, currency.options )
+							)
+						}
+					</FormSelect>
 				</Card>
 			</div>
 		);
 	}
 
 }
 
-export default localize( SettingsPaymentsLocationCurrency );
+function mapStateToProps( state ) {
+	const siteId = getSelectedSiteId( state );
+	const currency = getPaymentCurrencySettings( state, siteId );
+	return {
+		currency,
+		siteId,
+	};
+}
+
+function mapDispatchToProps( dispatch ) {
+	return bindActionCreators(
+		{
+			fetchSettingsGeneral
```

Comment:
I was thinking we would want to do the mapping for the fetch one level up, and just pass the currency in. Or is this the only setting from `general` that is going to be needed, and the rest are from other groups?